### PR TITLE
feat(query): purge files after insert with stage attachment

### PR DESCRIPTION
**Pull request by everpcpc**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_insert_v2.rs [hunk 1]
```diff
@@ -113,6 +117,30 @@ impl InsertInterpreterV2 {
...
+    async fn try_purge_files(
+        ctx: Arc<QueryContext>,
+        stage_info: &UserStageInfo,
+        stage_files: &[StageFileInfo],
+    ) {
+        let table_ctx: Arc<dyn TableContext> = ctx.clone();
+        let op = StageTable::get_op(stage_info);
+        match op {
+            Ok(op) => {
+                let file = Files::create(table_ctx, op);
```
Comment (ariesdevil) [inline src/query/service/src/interpreters/interpreter_insert_v2.rs:129]: I think the variable name change to `file_op` will be better.

**Review by ariesdevil — APPROVED** [review]

File: src/query/service/src/interpreters/interpreter_insert_v2.rs [hunk 2]
```diff
@@ -197,10 +228,18 @@ impl InsertInterpreterV2 {
                             start.elapsed().as_secs()
                         );
                         table
-                            .commit_insertion(ctx, append_entries, overwrite)
+                            .commit_insertion(ctx.clone(), append_entries, overwrite)
                             .await?;
 
-                        // TODO:(everpcpc) purge copied files
+                        if stage_info.copy_options.purge {
```
Comment (youngsofun) [inline src/query/service/src/interpreters/interpreter_insert_v2.rs:234]: only pure when success?

Comment (everpcpc) [reply src/query/service/src/interpreters/interpreter_insert_v2.rs:234]: this whole block seems all under success condition?

Comment (youngsofun) [reply src/query/service/src/interpreters/interpreter_insert_v2.rs:234]: yes, I means do we need co consider to pure when fail

**Review by youngsofun — APPROVED** [review]

**Comment by everpcpc** [comment]
@mergify update

**Pull request merged**